eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/AzureCollectionResultDefinition.cs

```diff
@@ -153,7 +153,12 @@ private MethodBodyStatement[] ConvertItemsToListOfBinaryData(VariableExpression
                     New.Instance(new CSharpType(typeof(List<>), typeof(BinaryData))), out itemsVariable),
                 new ForEachStatement("item", BuildGetPropertyExpression(Paging.ItemPropertySegments, responseVariable).As<IEnumerable<object>>(), out var itemVariable)
                 {
-                    itemsVariable.Invoke("Add", Static(typeof(ModelReaderWriter)).Invoke(nameof(ModelReaderWriter.Write), new ValueExpression[] { itemVariable, Static<ModelSerializationExtensionsDefinition>().Property("WireOptions") })).Terminate()
+                    itemsVariable.Invoke("Add", Static(typeof(ModelReaderWriter)).Invoke(nameof(ModelReaderWriter.Write),
+                        [
+                            itemVariable,
+                            Static<ModelSerializationExtensionsDefinition>().Property(ModelSerializationExtensionsDefinition.WireOptionsFieldName),
+                            Static<ModelReaderWriterContextDefinition>().Property("Default")
```

Comment:
Are we sure this will always exist?  It's possible for the local environment to turn off source generation, no?

Reply:
@jsquire I think 3p apps might turn off source generation but all of our sdks will have it on.

@JoshLove-msft right now we only have 1 serialization strategy which is MRW, if we get signal that people want to use this in a 3p setting with source generation off we will need to introduce a different serialization strategy like STJ attributes or something.  I don't think this is likely but just thinking through what the release valve might be.